### DISCUSSION TITLE
Smooth Scrolling in vim

### DISCUSSION THREAD

**Discussion by mikecgithub** [contour-terminal/contour#1307] [Ideas]
I always wanted to do this if I ever wrote a terminal, but I don't know how. 

When I open code in vim, and use ctrl-e or ctrl-y to scroll up/down, the text is unreadable because it jumps one line at a time. I don't think this is necessary, especially with GL acceleration. Sure, the CR jumps one line at a time, but once you know that, you can scroll the pix buff up at 30, or 60 fps, or whatever the user's resolution is. then the text will be as readable while scrolling as it is on a website.

Also, there is a bug in the documentation, http://contour-terminal.org/configuration/colors  this explains how to define color schemes, but it never says **how** to switch from one scheme to another. That should be the first thing in the docs under "color", in my opinion.

Thank you for all the hard work you have put into this project, I wish you the best!

**Comment by Yaraslaut** [comment]
Hi @mikecgithub

We can not change `vim` scroll behaviour, but we have `scroll_multiplier` configuration entry for scroll inside contour and maybe we can make our vi-mode scroll to adhere to this value, what do you think @christianparpart ?
   

Thanks for pointing our some issues with documentation. 
To specify  color scheme that you want to use for contour you need to specify color entry in the profile section of the configuration [colors](http://contour-terminal.org/configuration/profiles/#colors), maybe you can create a PR for our documentation to cover this topic? our documentation for colors is [here](https://github.com/contour-terminal/contour/blob/master/docs/configuration/colors.md)

**Comment by christianparpart** [comment]
> When I open code in vim, and use ctrl-e or ctrl-y to scroll up/down, the text is unreadable because it jumps one line at a time.

I think what you want is **actually** smooth scrolling by pixel, not by line, and that for a specific rectangular region within your text application, e.g. vim. I am so precise in saying that, because to date, no terminal emulator can do that.

Not yet at least, because in order to achieve this, a protocol extension would needed to be invented and implemented. Such a "VT extension" needs to be carefully designed, because it usually comes with a lot of corner cases and potential side effects (in this case more likely than others). And then the next challenge is, to get text applications to adapt to it, e.g. vim/neovim. And they are usually quite conservative and only implement new things if at least 2 or 3 (bigger) terminals already provide that feature, which is a classic chicken and egg problem, because the bigger terminal projects are usually also quite conservative.

I'm not saying this will never happening, but it's more likely that (if ever) it'll be projects like Contour, Foot, WezTerm, or Ghostty picking such new ideas up, and then only time will tell if TUI apps will follow up or not. We'll see.

On the other matter, our own `scroll_multiplier` feature, only applies to the history scrolling in primary screenl's, if a screen history does exist already.

@Yaraslaut instead of applying `scroll_multiplier` to vi normal mode (i have no strong preference on that matter, btw) - I just thought about adding `Ctrl+E` and `Scrol+Y` motions to vi normal mode instead. ;)

**Reply by Yaraslaut** [reply]
Indeed, we can add this motions
